scargo/file_generators/ut_gen.py

```diff
@@ -80,7 +80,10 @@ def _generate_cmake(self, src_dir_path: Path, ut_dir_path: Path) -> None:
         :param Path src_dir_path: Source directory for which tests are being generated
         :param Path ut_dir_path: Directory of generated unit tests
         """
-        add_subdirs_to_cmake(ut_dir_path.relative_to(self._project_path))
+        cmake_dir_path = ut_dir_path
+        while cmake_dir_path != self._ut_dir.parent:
+            add_subdirs_to_cmake(cmake_dir_path.relative_to(self._project_path))
```

Comment:
Does cmake really work like that, i.e. does it require adding the directory in all ancestor cmakelists? Can you provide some link to documentation that would confirm that?

Reply:
https://cmake.org/cmake/help/latest/command/add_subdirectory.html
In general, there are two possibilities for using cmake 1) have a single global cmake and include all files in the project (check GLOB), and have cmake in a more modular form in each directory. Then add_subdirectory() is needed on each folder level (our approach) or a relative path to the folder where another cmake is located is needed (e.g. add_subdirectory(my/relative/cmake/path)). In second solution you will not CMakeLists.txt file in each directory, but I think it is easier to manage the folder in this way

Reply:
Ok, I misunderstood, sorry